Add details about the kinds of problems that we work on (#41)
sourcegraph/careers: job-descriptions/software-engineer.md

### About the role

> You will help build Sourcegraph, a code search and navigation product, by collaborating on a small team to solve challenging problems that are fundamental to the growth and success of our business. Here are some examples of problems that we on working on every day:

Reviewer: The repeating "problems" wasn't good to my ear.

```suggestion
You will help build Sourcegraph, a code search and navigation product, by collaborating on a small team to solve challenging problems that are fundamental to the growth and success of our business. Here are some examples of what we work on every day:
```

- Create a code search experience that is friendly to new users and powerful in the hands of a veteran user.
- Scale code search to return results from tens of thousands of repositories and massive monorepos in under a second.
- Enable developers to perform large scale automated code modification.
- Provide code intelligence (e.g. hover tooltips, go to definition, find references) across repositories and in many languages.
- Integrate Sourcegraph features into existing code hosts (e.g. GitHub, Bitbucket, GitLab, Phabricator) via direct integrations and our browser extensions.
- Design an extension API that allows anyone to enhance their code browsing experience in Sourcegraph by surfacing relevant contextual information.

> - Guide customers to seamlessly deploy, configure, and upgrade Sourcegraph on-premisis in a variety of environments.

Reviewer 2: ```suggestion
- Guide customers to seamlessly deploy, configure, and upgrade Sourcegraph on-premise in a variety of environments.
```

- Automate release testing and continous deployments to our test clusters and sourcegraph.com while continuously monitoring important metrics and maintaining zero downtime.

We use a variety of technologies to help us accomplish our goals:

- Frontend: TypeScript, React, RxJS, SCSS, Browser extension APIs
- Backend: Go, GraphQL, PostgreSQL, Redis
- Deployment and operations: Kubernetes, Docker, Google Cloud Platform, Terraform

> We expect that you have two years of experience relevant to the problems that you will be working on and/or the technologies that you will be using at Sourcegraph.

Reviewer: ```suggestion
We expect that you have at least two years of experience relevant to the problems that you will be working on and/or the technologies that you will be using at Sourcegraph.
```
